packages/csharp/ArmoniK.Api.Client/Submitter/GrpcChannelFactory.cs

```diff
@@ -52,6 +55,150 @@ namespace ArmoniK.Api.Client.Submitter
   [PublicAPI]
   public static class GrpcChannelFactory
   {
+    /// <summary>
+    ///   Get the root certificates from the OS trusted store
+    /// </summary>
+    /// <returns>Root certificates in pem format</returns>
+    private static string GetRootCertificates()
+    {
+      var builder = new StringBuilder();
+      var store   = new X509Store(StoreName.Root);
+      store.Open(OpenFlags.ReadOnly);
+      foreach (var mCert in store.Certificates)
+      {
+        builder.AppendLine($"# Issuer: {mCert.Issuer}\n# Subject: {mCert.Subject}\n# Label: {mCert.FriendlyName}\n# Serial: {mCert.SerialNumber}\n# SHA1 Fingerprint: {mCert.GetCertHashString()}\n{ExportToPem(mCert)}\n");
+      }
+
+      store.Close();
+      return builder.ToString();
+    }
+
+    /// <summary>
+    ///   Exports a certificate to pem format
+    /// </summary>
+    /// <param name="cert">Certificate to export</param>
+    /// <returns>Certificate in pem formatted string</returns>
+    private static string ExportToPem(X509Certificate2 cert)
+      => $"-----BEGIN CERTIFICATE-----\n{Convert.ToBase64String(cert.GetRawCertData(), Base64FormattingOptions.InsertLineBreaks)}\n-----END CERTIFICATE-----";
```

Comment:
Is there no such method on a certificate object already?

Reply:
In net standard, afaik, no. https://stackoverflow.com/a/43941142